ngraph/core/src/op/util/convert_color_nv12_base.cpp

```diff
@@ -0,0 +1,231 @@
+        } else {
+            NODE_VALIDATION_CHECK(this,
+                                  uv_type.is_dynamic() || uv_type == y_type,
+                                  "UV input ",
+                                  uv_type,
+                                  " shall have same precision as Y input ",
+                                  y_type);
+        }
+        const auto& shape_uv = get_input_partial_shape(1);
+        NODE_VALIDATION_CHECK(this,
+                              shape_uv.rank().is_dynamic() || shape_uv.rank().get_length() == 4,
+                              "UV input with static shape shall have 4 dimensions (N, H, W, C)");
+        if (shape_y.rank().is_static() && shape_uv.rank().is_static()) {
+            // Verify that height for Y input is 2 times bigger than input height for UV
+            NODE_VALIDATION_CHECK(this,
+                                  shape_y[H_DIM].is_dynamic() || shape_uv[H_DIM].is_dynamic() ||
+                                      shape_y[H_DIM].get_length() == shape_uv[H_DIM].get_length() * 2,
+                                  "Y input height shall be 2 times bigger that UV input height: Y height = ",
+                                  shape_y[H_DIM].get_length(),
+                                  " UV height = ",
+                                  shape_uv[H_DIM].get_length());
+            // Verify that width for Y input is 2 times bigger than input width for UV
+            NODE_VALIDATION_CHECK(this,
+                                  shape_y[W_DIM].is_dynamic() || shape_uv[W_DIM].is_dynamic() ||
+                                      shape_y[W_DIM].get_length() == shape_uv[W_DIM].get_length() * 2,
+                                  "Y input width shall be 2 times bigger that UV input width: Y width = ",
+                                  shape_y[W_DIM].get_length(),
+                                  " UV width = ",
+                                  shape_uv[W_DIM].get_length());
+            NODE_VALIDATION_CHECK(this,
+                                  shape_uv[C_DIM].is_dynamic() || shape_uv[C_DIM].get_length() == 2,
+                                  "UV channels dimension shall be either dynamic or equal to 2. Current value is ",
+                                  shape_uv[C_DIM].get_length());
+
+            NODE_VALIDATION_CHECK(this,
+                                  shape_y[N_DIM].is_dynamic() || shape_uv[N_DIM].is_dynamic() ||
+                                      shape_y[N_DIM].get_length() == shape_uv[N_DIM].get_length(),
+                                  "Y input batch shall be same as UV input batch: Y batch = ",
+                                  shape_y[N_DIM].get_length(),
+                                  " UV batch = ",
+                                  shape_uv[N_DIM].get_length());
+        }
+        // Set shape based on UV shape, if Y are dynamic
+        if (shape_uv.rank().is_static()) {
+            if (out_shape[N_DIM].is_dynamic()) {
```

Comment:
why do we need to check that?

Reply:
If batch for Y is not defined, but batch for UV is defined - then use batch value from UV

Reply:
we should use ngraph::Dimension::merge